docusaurus/docs/Android/02-client/06-guides/08-optimizations.mdx

```diff
@@ -22,3 +22,11 @@ Call<T>.forceNewRequest(): Call<T>
 The returned `Call` will be forced to make a new request to API. 
 
 
+## QuerySorter
+To sort your object and present then in a desired order, tt is possible to choose between 2 implementations of QuerySorter: `QuerySort` and `QuerySortByMap`. 
+
+`QuerySort` uses reflection to find the fields that should be used to compare when sorting. This avoids the need for extra implementation to make a class able to be sorted. 
+
+The drawback is that reflection is an expensive operation and this implementation will have a lower performance compared to `QuerySortByMap`.
+
+On the other hand, `QuerySortByMap` needs the class to be sorted to extend from `QueryableByMap` and provide the fields to be used in the comparation in a map. This implementation doesn't use reflection, so it will run faster. 
```

Comment:
```suggestion
You can also choose to not use reflection and gain performance by using `QuerySortByMap` instead but this requires some extra work. Your custom class needs to implement the `QueryableByMap` interface and provide the names of the fields used to sort the list.
```

Reply:
Let's also add something like this below:
```
:::note
Models included in the SDK already implement the `QueryableByMap` interface and include all the fields together with `extraData`!
:::
```

Reply:
Can we also add an example of how to implement it? Maybe we can copy Channel, Message or any other model?